roles/libvirt_manager/tasks/manage_vms.yml

```diff
@@ -1,44 +1,37 @@
 ---
-- name: "(localhost) Inject ssh jumpers for {{ vm }}"
-  when:
-    - inventory_hostname != 'localhost'
-    - inventory_hostname != 'instance'  # needed for molecule
-  delegate_to: localhost
-  vars:
-    proxy_hostname: "{{ ansible_host | default(inventory_hostname) }}"
-  ansible.builtin.blockinfile:
-    create: true
-    path: "{{ lookup('env', 'HOME') }}/.ssh/config"
-    marker: "## {mark} {{ vm }} {{ inventory_hostname }}"
-    mode: "0600"
-    block: |-
-      Host {{ vm }} {{ vm }}.{{ inventory_hostname }} cifmw-{{ vm }} {{ extracted_ip }}
-        ProxyJump {{ ansible_user | default(lookup('env', 'USER')) }}@{{ proxy_hostname }}
-        Hostname {{ extracted_ip }}
-        User {{ 'core' if vm is match('^(crc|ocp).*') else 'zuul' }}
-        StrictHostKeyChecking no
-        UserKnownHostsFile /dev/null
-
-- name: "({{ inventory_hostname }}) Inject ssh jumpers for {{ vm }}"  # noqa: name[template]
+- name: "Push ssh jumper/configuration for {{ vm }}"
   vars:
-    identity_file: >-
-      {{
-        cifmw_libvirt_manager_basedir ~ '/artifacts/cifmw_ocp_access_key' if vm_type is match('^ocp.*') else
-        ansible_user_dir ~ '/.crc/machines/crc/id_ecdsa' if vm_type == 'crc' else
-        ansible_user_dir ~ '/.ssh/cifmw_reproducer_key'
-      }}
-  ansible.builtin.blockinfile:
-    create: true
-    path: "{{ ansible_user_dir }}/.ssh/config"
-    marker: "## {mark} {{ vm }}"
-    mode: "0600"
-    block: |-
-      Host {{ vm }} {{ vm }}.{{ inventory_hostname }} cifmw-{{ vm }} {{ extracted_ip }}
-        Hostname {{ extracted_ip }}
-        User {{ 'core' if vm is match('^(crc|ocp)') else 'zuul' }}
-        IdentityFile {{ identity_file }}
-        StrictHostKeyChecking no
-        UserKnownHostsFile /dev/null
+    _user: "{{ 'core' if vm is match('^(crc|ocp).*') else 'zuul' }}"
+    dataset:
+      ssh_dir: "{{ ansible_user_dir }}/.ssh"
+      user: "{{ _user }}"
+      hostname: "{{ extracted_ip }}"
+      patterns:
+        - "{{ vm }}"
+        - "{{ vm }}.{{ inventory_hostname }}"
+        - "cifmw-{{ vm }}"
```

Comment:
We are removing the ip address from patterns, I guess that is ok?

Reply:
I checked the role, it appends the "hostname" value:
https://github.com/openstack-k8s-operators/ci-framework/blob/c6806319c0620e34df502fd2e78ae8235edd6cdf/roles/ssh_jumper/templates/ssh_host.conf.j2#L1

Reply:
ah, right - I did that. :rofl: All good. :+1: